import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/CStoreService.java

```diff
@@ -75,15 +79,29 @@ protected void store(
       validateParam(sopClassUID, "AffectedSOPClassUID");
       validateParam(sopInstanceUID, "AffectedSOPInstanceUID");
 
-      CountingInputStream countingStream = new CountingInputStream(inDicomStream);
-      InputStream inBuffer =
+      CountingInputStream countingStream;
+      IDicomWebClient destinationClient;
+      if(destinationMap != null && destinationMap.size() > 0){
+        DicomInputStream inDicomStream  = new DicomInputStream(inPdvStream);
+        inDicomStream.mark(Integer.MAX_VALUE); // do or die (OOM)
```

Comment:
In theory this could get out of memory, since mark parameter sets size limit for BufferedInputStream inside the DicomInputStream.
But it's unlikely unless instance has unreasonable amount of OB (other binary) tags or sequences.
I think it's safe to remove this comment.